src/local_spawn_handle.rs

```diff
@@ -20,7 +20,7 @@ pub trait LocalSpawnHandle<Out: 'static>
 }
 
 
-/// Let's you spawn a !Send future and get a [JoinHandle] to await the output of a future.
+/// Let you spawn a !Send future and get a [JoinHandle] to await the output of a future.
```

Comment:
Actually the correct one is "Lets".

Reply:
> There seems to be also an issue with examples/spawn_handle_multi.rs. It looks like the `JoinHandle` now has `NonNull` in it which isn't `Send`. Not sure yet what the best solution is.


Yes I also find out, which is bad

```rust
error[E0277]: `NonNull<()>` cannot be sent between threads safely
   --> async_executors/src/core/join_handle.rs:262:2
    |
262 |     static_assertions::assert_impl_all!(JoinHandle<()>: Send);
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |     |
    |     `NonNull<()>` cannot be sent between threads safely
    |     required by this bound in `assert_impl_all`
    |
    = help: within `join_handle::JoinHandle<()>`, the trait `std::marker::Send` is not implemented for `NonNull<()>`
    = note: required because it appears within the type `glommio::task::JoinHandle<()>`
    = note: required because it appears within the type `Option<glommio::task::JoinHandle<()>>`
    = note: required because it appears within the type `glommio::executor::multitask::Task<()>`
    = note: required because it appears within the type `glommio::Task<()>`
    = note: required because it appears within the type `Option<glommio::Task<()>>`

```

Reply:
This seems to refer to Glommio's Task JoinHandle. Because Glommio uses a thread-per-core design, it leverages a lot of optimizations that eliminates atomics in the task's state machine. That means that it can never escape the executor in which it was created. So It indeed cannot be `Send`.

Reply:
So the choices are:
1. Go back to RemoteHandle at cost of performance drop
2. Split `JoinHandle` to `JoinHandle(:Send)` and `LocalJoinHandle(:!Send)` at the cost of a beaking change.

Reply:
I feel the little bit of perf we would get out of this isn't worth breaking up the API. Also, even when using a local executor, it can be handy and desirable to await the outcome on another thread when `Out` is `Send`. If we make `LocalSpawnHandle` return a `!Send` `JoinHandle`, we not only affect `glommio`, but also other executors which don't have this problem.

So I would say let's do `RemoteHandle` for this. The overhead is just a matter of a oneshot channel.

Reply:
Should all task bahave like "if we drop it without `detach`ing, it will be cancelled"? I think RemoteHandle cannot cancel tasks at least.

Reply:
Yes, all have consistent behavior. That's what most of the code in joinhandle.rs does as tokio and async-std reverse this. `RemoteHandle` already does this out of the box. But all we need is to use `JoinHandle::RemoteHandle`. All the code is already there for detaching and dropping.